web/views.py

```
@@ -5242,6 +5243,167 @@ def toggle_course_status(request, slug):
+        try:
+            wallet_data = create_student_wallet()  # Now generates a mnemonic
+            # Store only public address
+            profile, created = Profile.objects.get_or_create(user=student)
+            profile.wallet_address = wallet_data["address"]
+            profile.save()
+            # Wallet address successfully saved to profile
+            # Store mnemonic temporarily in session (Do NOT store in DB)
+            request.session["mnemonic"] = wallet_data["mnemonic"]
+            request.session["public_key"] = wallet_data["address"]
+            return redirect("show_mnemonic")  # Redirect to mnemonic page
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)